pkg/virtctl/convert-machine-type/convert-machine-type-job/update_machine_type.go

```diff
@@ -0,0 +1,152 @@
+package convertmachinetypejob
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	k8sv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k6tv1 "kubevirt.io/api/core/v1"
+)
+
+// using these as consts allows us to easily modify the program to update as newer versions are released
+// we generally want to be updating the machine types to the most recent version
+const (
+	LatestMachineTypeVersion           = "rhel9.2.0"
+	MinimumSupportedMachineTypeVersion = "rhel9.0.0"
+)
+
+var (
+	// by default, update machine type across all namespaces
+	namespace     = k8sv1.NamespaceAll
+	labelSelector = ""
+
+	// by default, should require manual restarting of VMIs
+	restartNow = false
+)
+
+func IsMachineTypeUpdated(machineType string) (isUpdated bool, updatedMachineType string, err error) {
+
+	// if the machine type is "q35", we want the VM to behave as though its machine type has been updated
+	if machineType == "q35" {
+		return true, machineType, nil
+	}
+
+	machineTypeSubstrings := strings.Split(machineType, "-")
```

Comment:
We make the assumption of "-" being a delimiter, but some machine types do not follow it. I.e pc and q35.